### DISCUSSION TITLE
Configureable discovery topic name?

### DISCUSSION THREAD

**Discussion by igorbernstein** [arendst/Tasmota#11060] [Ideas]
Hi,

I was wondering if you would consider taking a slightly different approach for home assistant discovery config naming? 
I would like to lock down the acls in my MQTT broker so that each device is restricted to its own namespace. Ideally I would like to use predictable patterns to define the acls. Something like:

pattern read cmnd/tasmotas/+
pattern cmnd/%u/+
pattern tele/%u/+
pattern stat/%u/+
pattern tasmota/discovery/%u/config

Where %u is the mqtt user. To accomplish this, I'm currently forced to use the Mac id as the user name since the discovery topic hardcoded here:
https://github.com/arendst/Tasmota/blob/d5542b1a66322d4abc99792d56f7ff98bbb11249/tasmota/xdrv_12_home_assistant.ino#L334

This is a bit unfortunate because it makes it hard for me troubleshoot the topics since the Mac ids are meaningless to a human. Alternatively, I can use a friendlier user name, but then I would have to define an explicit acl for the discovery topic.

From what I can tell the Mac id in the topic is not actually used by anything, so I was wondering if it can be made configureable? 

Thanks in advance!

**Comment by ascillato** [comment]
The discovery topic structure is a standard for home assistant. Changing that will make the Tasmota HA Native integration to not work. Sorry, that won't be implemented.
